kyokan/bob-wallet
# Reserved Name Claims

This adds reserved name claim functionality to the UI.

~DO NOT MERGE until commit deca1a5f1a6c3076560e71194d58046595d254ff is removed~ ✅  OK

## TODO:
- [x] Add Method 3: sign offline and paste RRSIG
- [x] Add Method 4: sign offline using PKCS11 for TLD owners who manage with OpenDNSSEC + HSM
- [x] Add "Success" modal when Claim is broadcast (note that claims do not appear in the mempool!)
- [x] MAKE IT PRETTY (@rithvikvibhu and @chikeichan have kindly offered to polish the UI for me)

## To test:

This branch uses a hard-fork of hsd to test, which includes two extra reserved names `hns-claim-test-1.xyz` and `hns-claim-test-2.xyz`, which are two real ICANN domains that I own and set up for this purpose. When you launch Bob using `npm run dev` it will automatically start in regtest mode and generate a few blocks, including the block I used in the claim commitment. 

To claim `hns-claim-test-1` you must create a wallet with my favorite seed phrase of all time:

```
abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon about
```

...and then you can use "Method 1" which is the easiest, since I have already pasted the claim TXT in "the" DNS:

```
$ dig hns-claim-test-1.xyz txt +short
"hns-regtest:aakkrwicqqs2s5aoxavbcaariycxuze3i5fp3ub6n2r62zx3tjtl3ry56orjwonppjdru3p2uirlbyglvvtcepoqbhdacaaaadjczi34"
```

To claim `hns-claim-test-2` you need my DNSSEC private key on your own computer which you then use to sign with "Method 2". That private key is posted [here](https://gist.github.com/pinheadmz/4712b2db28d14d0071ed30868f0410ef). Note that the filename on your local machine must match this filename exactly: `Khns-claim-test-2.xyz.+008+27259.private`

## Navigating

The reserved name claim page is currently accessible in two ways: By looking up a reserved name, or by clicking "Airdrop / Name Claim" in the sidebar, then "Claim":

<img width="999" alt="Screen Shot 2021-06-02 at 2 00 45 PM" src="https://user-images.githubusercontent.com/2084648/120529987-01166180-c3ab-11eb-8263-bccd7b39751e.png">
<img width="317" alt="Screen Shot 2021-06-02 at 2 00 30 PM" src="https://user-images.githubusercontent.com/2084648/120529989-01aef800-c3ab-11eb-8603-a4d14d2296b7.png">

## Usage

<img width="1024" alt="Screen Shot 2021-06-02 at 2 01 54 PM" src="https://user-images.githubusercontent.com/2084648/120530084-1ee3c680-c3ab-11eb-84dc-422507c90008.png">
